subsys/net/l2/ppp/ppp_l2.c

```diff
@@ -457,4 +496,34 @@ void net_ppp_init(struct net_if *iface)
 
 	k_work_reschedule(&ctx->startup,
 			  K_MSEC(CONFIG_NET_L2_PPP_DELAY_STARTUP_MS));
+
+	if (!tx_init) {
+		k_tid_t tid;
+
+		tx_init = true;
+
+		k_fifo_init(&tx_queue);
+
+		tid = k_thread_create(&tx_handler_thread, tx_handler_stack,
+				      K_KERNEL_STACK_SIZEOF(tx_handler_stack),
+				      (k_thread_entry_t)tx_handler,
+				      NULL, NULL, NULL,
+				      THREAD_PRIORITY, 0, K_FOREVER);
+		if (!tid) {
+			NET_ERR("Cannot create TX handler thread for %d",
+				net_if_get_by_iface(iface));
+		} else {
+			if (IS_ENABLED(CONFIG_THREAD_NAME)) {
+#define MAX_NAME_LEN sizeof("ppp_tx[01]")
+				char name[MAX_NAME_LEN];
+
+				snprintk(name, sizeof(name), "ppp_tx[%d]",
+					 net_if_get_by_iface(iface));
```

Comment:
There will be only single tx thread, so there is no need to add interface number to thread name.

Reply:
> There will be only single tx thread, so there is no need to add interface number to thread name.

Sure, I can tweak this.